docs/multibrokers.md

```diff
@@ -0,0 +1,218 @@
+
+`writeMessageHandler`:
+
+```go
+func (s *Server) writeMessageHandler() rest.Handler {
+	return func(w http.ResponseWriter, r *http.Request) {
+		logger.Info("handling message write")
+			
+        // From the request's URL the LB obtains the targeted channel.
+			channel := strings.TrimPrefix(r.URL.Path, "/")
+        
+		// Verification step to make sure the node can output on the targeted channel
+            if !environment.OutputChannelList().Contains(channel) {
+                logger.Error(fmt.Sprintf("channel %s not found in output channel list", channel))
+                insprError := ierrors.NewError().
+                    BadRequest().
+                    Message("channel '%s' not found", channel)
+
+                rest.ERROR(w, insprError.Build())
+                return
+            }
+		
+        // Once the channel is verified the broker searches for it's respective broker
+            channelBroker, err := environment.GetChannelBroker(channel)
+            if err != nil {
+                logger.Error("unable to get channel broker",
+                    zap.String("channel", channel),
+                    zap.Any("error", err))
+
+                rest.ERROR(w, err)
+                return
+            }
+
+        // With all the channel's information parsed and verified, LB fetches the information 
+        //required to reach the correct broker specific sidecar
+            sidecarAddress := environment.GetBrokerSpecificSidecarAddr(channelBroker)
+            sidecarWritePort := environment.GetBrokerWritePort(channelBroker)
+
+			reqAddress := fmt.Sprintf("%s:%s/%s", sidecarAddress, sidecarWritePort, channel)
+
+		logger.Debug("encoding message to Avro schema")
+        
+		// Message is encoded to avro
+            encodedMsg, err := encodeToAvro(channel, r.Body)
+            if err != nil {
+                logger.Error("unable to encode message to Avro schema",
+                    zap.String("channel", channel),
+                    zap.Any("error", err))
+
+                rest.ERROR(w, err)
+                return
+            }
+
+		logger.Info("sending message to broker",
+			zap.String("broker", channelBroker),
+			zap.String("channel", channel))
+
+        // Forwards the message to the broker specific sidecar
+            resp, err := sendRequest(reqAddress, encodedMsg)
+            if err != nil {
+                logger.Error("unable to send request to "+channelBroker+" sidecar",
+                    zap.Any("error", err))
+
+                rest.ERROR(w, err)
+                return
+            }
+		
+        // Responds the Node
+			rest.JSON(w, resp.StatusCode, resp.Body)
+	}
+}
```

Comment:
formatting, no reason for the example code to not be formatted.

Reply:
It is formatted for better readability